k8s_test_harness/util/k8s_util.py

```diff
@@ -114,3 +123,73 @@ def get_join_token(
 # Join an existing cluster.
 def join_cluster(instance: harness.Instance, join_token: str):
     instance.exec(["k8s", "join-cluster", join_token])
+
+
+def get_helm_command(
+    name: str,
+    chart_name: str,
+    namespace: str = constants.K8S_NS_KUBE_SYSTEM,
+    repository: str = None,
+    images: List[HelmImage] = None,
+    set_configs: List[str] = None,
+):
+    """Creates a helm install command for the given helm chart.
+
+    This function expects that the variables set in the given images can be
+    found as environment variables.
+
+    This function will prepend a "--set" string before each element in
+    set_configs in the final helm install command.
+
+    Returns:
+        list of strings representing the helm install command.
+    """
+    images = images or []
+    set_configs = set_configs or []
+
+    helm_command = [
+        "k8s",
+        "helm",
+        "install",
+        name,
+        chart_name,
+        "--namespace",
+        namespace,
+        "--create-namespace",
+    ]
+
+    if repository:
+        helm_command += [
+            "--repo",
+            repository,
+        ]
+
+    for image in images:
+        image_uri = os.getenv(image.variable)
+        assert image_uri is not None, f"{image.variable} is not set"
+        image_split = image_uri.split(":")
+        image_name = image_split[0]
+
+        # This helm charts requires setting the image registry separately.
+        parts = image_name.split("/")
+        if len(parts) > 1:
+            image_name = "/".join(parts[1:])
+
+        prefix = ""
+        if image.prefix:
+            prefix = f"{image.prefix}."
+
+        helm_command += [
+            "--set",
+            f"{prefix}image.repository={image_name}",
+            "--set",
+            f"{prefix}image.tag={image_split[1]}",
+            "--set",
+            "{prefix}securityContext.runAsUser=584792",
```

Comment:
I'd make this configurable with sane default

Reply:
Done